source/client-side-operations-timeout/client-side-operations-timeout.rst

```diff
@@ -0,0 +1,927 @@
+
+Cursors
+-------
+
+For operations that create cursors, ``timeoutMS`` can either cap the lifetime
+of the cursor or be applied separately to the original operation and all
+``next`` calls. To support both of these use cases, these operations MUST
+support a ``timeoutMode`` option. This option is an enum with possible values
+``CURSOR_LIFETIME`` and ``ITERATION``. The default value depends on the type
+of cursor being created. Drivers MUST error if ``timeoutMode`` is set and
+``timeoutMS`` is not.
+
+When applying the ``timeoutMS`` option to ``next`` calls on cursors, drivers
+MUST ensure it applies to the entire call, not individual commands. For
+drivers that send ``getMore`` requests in a loop when iterating tailable
+cursors, the timeout MUST apply to the totality of all ``getMore``’s, not to
+each one individually. If a resume is required for a ``next`` call on a
+change stream, the timeout MUST apply to the entirety of the initial
+``getMore`` and all commands sent as part of the resume attempt.
+
+For ``close`` methods, drivers MUST allow ``timeoutMS`` to be overridden if
+doing so is possible in the language.  If explicitly set for the operation,
+it MUST be honored.  Otherwise, if ``timeoutMS`` was applied to the operation
+that created the cursor, it MUST be refreshed for the ``killCursors`` command
+if one is required.  Note that this means ``timeoutMS`` will be refreshed for
+the ``close`` call even if the cursor was created with a ``timeoutMode`` of
+``CURSOR_LIFETIME`` and the timeout associated with the cursor has expired.
+The calculated timeout MUST apply to explicit ``close`` methods that can be
+invoked by users as well as implicit destructors that are automatically
+invoked when exiting resource blocks.
+
+See `Cursor close() methods refresh timeoutMS`_.
+
+Non-tailable Cursors
+~~~~~~~~~~~~~~~~~~~~
+
+For non-tailable cursors, the default value of ``timeoutMode`` is
+``CURSOR_LIFETIME``. If ``timeoutMS`` is set, drivers MUST apply it to the
+original operation and the lifetime of the created cursor. For example, if a
+``find`` is executed at time ``T``, the ``find`` and all ``getMore``’s on the
+cursor must finish by time ``T + timeoutMS``. When executing ``next`` calls
+on the cursor, drivers MUST use the remaining timeout as the ``timeoutMS``
+value for the operation but MUST NOT append a ``maxTimeMS`` field to
+``getMore`` commands. If there are documents remaining in a previously
+retrieved batch, the ``next`` method MUST return them even if the timeout has
+expired and MUST only return a timeout error if a ``getMore`` is required.
+
+If ``timeoutMode`` is set to ``ITERATION``, drivers MUST first validate the
+operation and return a client-side error if the operation is an ``aggregate``
+with a ``$out`` or ``$merge`` pipeline stage.  If the operation is valid,
+drivers MUST honor the ``timeoutMS`` option for the initial command but MUST
+NOT append a ``maxTimeMS`` field to the command sent to the server.  After
```

Comment:
AFAICT, "validation" here is only referring to whether the operation is an aggregate with `$out`/`$merge`. Since "valid" can have other meanings when we're talking about operations (e.g. update/replace arguments) I suggest rephrasing for clarity:

> If ``timeoutMode`` is set to ``ITERATION``, drivers MUST raise a client-side error if the operation is an ``aggregate``
with an ``$out`` or ``$merge`` pipeline stage.  If the operation is not an  ``aggregate`` with ``$out`` or ``$merge``, drivers MUST honor the ``timeoutMS`` option for the initial command but MUST NOT append a ``maxTimeMS`` field to the command sent to the server.

Side nit: this paragraph seems to use two spaces after a period, which is inconsistent with other paragraphsi n the spec.

Reply:
Fixed to use the suggested rewording and remove the extra spaces.